addons/confirma/src/exceptions/ConfirmAssertException.cs

```diff
@@ -9,5 +11,41 @@ public ConfirmAssertException(string message) : base(message) { }
     public ConfirmAssertException(string message, Exception innerException)
     : base(message, innerException) { }
 
+    public ConfirmAssertException(
+        string assertion,
+        object expected,
+        object actual
+    )
+    : base(
+        new AssertionMessageGenerator(
+            Formatter.DefaultFormat,
+            assertion,
+            new DefaultFormatter(),
+            expected,
+            actual
+        ).GenerateMessage()
+    )
+    { }
+
+    public ConfirmAssertException(
+        string format,
+        string assertion,
+        Formatter? formatter,
+        object? actual,
+        object? expected,
+        string? customMessage
+    )
+    : base(
+        customMessage
+        ?? new AssertionMessageGenerator(
```

Comment:
I see no reason to duplicate a `ConfirmAssertException(string Message)` with a `customMessage`

Reply:
This might not be the best implementation, but actually has a good use case.

Let's take a look at `ConfirmCompletesWithin` extension. Now it is possible to write it like this:

```cs
    public static Action ConfirmCompletesWithin(
        this Action action,
        TimeSpan timeSpan,
        string? message = null
    )
    {
        return !Task.Run(action).Wait(timeSpan)
            ? throw new ConfirmAssertException(
                "Expected action to complete within {1} ms, but it timed out.",
                nameof(ConfirmCompletesWithin),
                new NumericFormatter(2),
                null,
                timeSpan.TotalMilliseconds,
                message
            )
            : action;
    }
```

But without `customMessage` it's impossible and must be written like this (or something similar):

```cs
    public static Action ConfirmCompletesWithin(
        this Action action,
        TimeSpan timeSpan,
        string? message = null
    )
    {
        if (!Task.Run(action).Wait(timeSpan))
        {
            return action;
        }
        
        if (message is not null)
        {
            throw new ConfirmAssertException(message);
        }

        throw new ConfirmAssertException(
            "Expected action to complete within {1} ms, but it timed out.",
            nameof(ConfirmCompletesWithin),
            new NumericFormatter(2),
            null,
            timeSpan.TotalMilliseconds,
            message
        );
    }
```

So as you can see this is quite useful, but I understand concerns. If you know a way to achieve the same effect with a better approach, I gladly use it.

Reply:
yep, that makes sens now